src/logic/graphFrames.ts

```diff
@@ -1,19 +1,95 @@
-import { Frame, DissasembledGraph, Algorithm } from './commonInterfaces';
+import {
+  Frame,
+  DisassembledGraph,
+  Algorithm,
+  Edge,
+  TraversalRecord,
+  Paths,
+  GraphIndex,
+} from './commonInterfaces';
+
+interface FrameWithEdges {
+  activeEdges: Array<Edge>,
+  inactiveEdges: Array<Edge>,
+};
 
 export default function createFrames(
-  dissasembledGraph: DissasembledGraph,
+  dissasembledGraph: DisassembledGraph,
   algorithm: Algorithm,
 ):
   Array<Frame>
 {
-  return [ // Temporary fixed result
-    { //frame 1
-      inactiveEdges: [],
-      activeEdges: [dissasembledGraph[1][0]],
-    },
-    { //frame 2
-      inactiveEdges: [dissasembledGraph[1][0]],
-      activeEdges: [],
-    },
-  ]
+  const edgeFrames = createFramesWithEdges(algorithm);
+  return convertToIndexes(dissasembledGraph, edgeFrames);
+}
+
+export function createFramesWithEdges(from: Algorithm) {
+  let frames: Array<FrameWithEdges> = [];
+  for (let tr of from.traversalHistory) {
+    frames.push(createFrameForNode(frames));
+    frames.push(...createFramesForLookups(tr, frames));
+  }
+  return frames;
+}
+
+function createFrameForNode(frames: Array<FrameWithEdges>): FrameWithEdges {
+  const lastFrame = frames[frames.length - 1] || { activeEdges: [], inactiveEdges: [] };
+  return {
+    activeEdges: [],
+    inactiveEdges: lastFrame.activeEdges.concat(lastFrame.inactiveEdges),
+  };
+}
+
+function createFramesForLookups(
+  tr: TraversalRecord,
+  frames: Array<FrameWithEdges>,
+):
+  Array<FrameWithEdges>
+{
+  const nodeFrames: Array<FrameWithEdges> = [
+    frames[frames.length - 1] || { activeEdges: [], inactiveEdges: [] },
+  ];
+  for (let toNode of tr.lookups) {
+    const lastFrame = nodeFrames[nodeFrames.length - 1];
+    nodeFrames.push({
+      activeEdges: [new Edge(tr.node, toNode)],
+      inactiveEdges: lastFrame.activeEdges.concat(lastFrame.inactiveEdges),
+    });
+  }
+  nodeFrames.shift() //First frame was repeated
```

Comment:
so maybe let's cut the array earlier ?

Reply:
Please explain

Reply:
Instead of adding a difficult-to-understand line (every line with the comment is automatically harder to analyse), let's find out the solution to prevent it at earlier stage

Reply:
That much I understood. I was rather asking how/

Reply:
Maybe let's start the iteration from second elem or use Set